SPLASH/src/game/states/GameState.cpp

```diff
@@ -926,6 +930,34 @@ void GameState::updatePerTickKillCamComponentSystems(float dt) {
 // HERE BE DRAGONS
 // Make sure things are updated in the correct order or things will behave strangely
 void GameState::updatePerTickComponentSystems(float dt) {
+	
+	m_playerNamesinGameGui.clearPlayersToDraw();
+	if (!m_player->getComponent<SpectatorComponent>()) {
+		m_playerNamesinGameGui.setMaxDistance(10);
+
+		Octree::RayIntersectionInfo tempInfo;
+		m_octree->getRayIntersection(m_cam.getPosition(), m_cam.getDirection(), &tempInfo, m_player, 0.0, true);
+		if (tempInfo.closestHitIndex != -1) {
+			NetworkReceiverComponent* nrc = tempInfo.info[tempInfo.closestHitIndex].entity->getComponent<NetworkReceiverComponent>();			
+			if (nrc) {
+				CandleComponent* candleComp = tempInfo.info[tempInfo.closestHitIndex].entity->getComponent<CandleComponent>();
+
+				Netcode::PlayerID owner = Netcode::getComponentOwner(nrc->m_id);
+				Player* p = NWrapperSingleton::getInstance().getPlayer(owner);
+
+				if (p) {
+					m_playerNamesinGameGui.addPlayerToDraw(tempInfo.info[tempInfo.closestHitIndex].entity);
+				}
+			}
+		}
+	} else {
+		m_playerNamesinGameGui.setMaxDistance(-1);
+		for (auto p : *m_componentSystems.playerSystem->getPlayers()) {
+			m_playerNamesinGameGui.addPlayerToDraw(p);
+		}
+	}
+	
+	///////////////////////////////////////
```

Comment:
THis could probably be done somewhere else. Because it only affects readability I think its ok at this stage.